ftw/showroom/js/test/spec/showroom.js

```diff
@@ -438,5 +422,51 @@ describe("Showroom", () => {
 
   });
 
+  describe("throttling", () => {
+    it("should not render more than one item within 250ms when hitting right arrow several times", (done) => {
+      fixture.load("default_outlet.html");
+
+      let renderCalls = 0;
+      let showroom = Showroom(defaultItems, {
+        fetch: () => { return "<div></div>" },
+        render: () => { renderCalls += 1 }
+      });
+
+      event.hitArrowRight(fixture.el.querySelector("#outlet"));
+      setTimeout(() => {
+        event.hitArrowRight(fixture.el.querySelector("#outlet"));
+        assert.equal(renderCalls, 1, "The render method should have been called only one within 1000ms");
+        done();
+      }, 950);
```

Comment:
This might lead to fluky behavior, i'd suggest to replace this with mocking, if possible. Otherwise at least reduce the timeout to avoid too much waiting time 😉 .